File: katas/Palindrom/solutions/jonas/Palindrom/Palindrom.Test/PalindromTests.cs
using NUnit.Framework;
using Palindrom.TestAdapter;

namespace Palindrom.Test
{
    public class PalindromTests
    {
        private IPalindromTestAdapterObject _palindromTestAdapterObject;
        [SetUp]
        public void Setup()
        {
            _palindromTestAdapterObject = PalindromTestAdapterFactory.Create();
        }

        [Test]
        [TestCase("abba", true)]
        [TestCase("Reliefpfeiler", true)]
        [TestCase("Rentner", true)]
        [TestCase("Fumpp", false)]
        [TestCase("Kein Palindrom", false)]
        [TestCase("Dienstmannamtsneid", true)]
        [TestCase("Tarne nie deinen Rat!", true)]
        [TestCase("Eine güldne, gute Tugend: Lüge nie!", true)]
        [TestCase("Ein agiler Hit reizt sie. Geist?! Biertrunk nur treibt sie. Geist ziert ihre Liga nie!", true)]
        public void Palindrom_Test(string inputString, bool isPalindrom)
        {
            var result = _palindromTestAdapterObject.IsPalindrom(inputString);
            Assert.AreEqual(isPalindrom, result);
        }
    }
}

File: katas/Palindrom/solutions/1001binary/palindrom_tests.py
import unittest
import palindrom
from parameterized import parameterized

class TestPalindrom(unittest.TestCase):
    @parameterized.expand([
        ["abba"],
        ["rentner"],
        ["Lagerregal"],
        ["Dienstmannamtsneid"],
        ["Eine güldne, gute Tugend: Lüge nie!"],
        ["Tarne nie deinen Rat!"],
        ["Ein agiler Hit reizt sie. Geist?! Biertrunk nur treibt sie. Geist ziert ihre Liga nie!"]
    ])
    def testpalindrom_text_returnstrue(self, text):
        self.assertEqual(True, palindrom(text))

    @parameterized.expand([
        [None],
        [""],
        ["   "]
    ])
    def testpalindrom_text_returnsfalse(self, text):
        self.assertEqual(False, palindrom(text))

if __name__ == '__main__':
    unittest.main()